network/tcp_socket.c

```diff
@@ -339,7 +353,21 @@ int32_t socket_connect(struct tcp_socket_desc *desc,
 					desc->id, addr);
 	if (NO_OS_IS_ERR_VALUE(ret))
 		return ret;
-
+#ifdef NO_OS_LWIP_NETWORKING
+		/*
+		 * Currently, the LWIP networking layer doesn't implement packet RX
+		 * using interrupts, so we have to poll.
+		 * 
+		 * Adding this as a temporary workaround, based off this commit: https://github.com/analogdevicesinc/no-OS/commit/8ca2a15b8b7cc7a53985bf5a773bdd99f4bc63bc#diff-f9781ad4653f9192f0521846b4116a5b34ae3c2abd2c4256ee0a813e3f947aa4R158
+		 * 
+		 * Previously this polling was only applied to the mqtt layer
+		 */
+	int i = 500;
+	while(i>0){
+		no_os_lwip_step(desc->net->net, NULL);
+		i--;
+	}
+#endif /*NO_OS_LWIP_NETWORKING*/
```

Comment:
Instead of doing this, I'd say we move this sequence after: https://github.com/analogdevicesinc/no-OS/blob/main/network/lwip_raw_socket/lwip_socket.c#L876

Also, you'll need to add a delay (1ms for example) between `no_os_lwip_step()` calls. This way we could configure the timeout.

Reply:
Thank you, I've now used the no_os_delay method and moved this sequence to the lwip_socket_connect function. My project still builds and runs correctly with the change